Imtiaz branch

tiazahmd opened the pull request:
Left space and complexity analysis as to-do since I don't know how to do them yet.

MrHeer commented on src/PalindromeNumber/README.md:
```diff
@@ -0,0 +1,30 @@
+1. If it's a single digit, then it's a Palindrome Number. 
+2. Ignore any negative integers.
+3. For all other types of integers, pop each digit and store them in a vector in order.
+4. Compare the `(n + i)th` integer with `(vector.size() - 1 - i)th integer. 
```
```suggestion
4. Compare the `(n + i)th` integer with `(vector.size() - 1 - i)th` integer. 
```

MrHeer commented on src/PalindromeNumber/Solution.cpp:
```diff
@@ -0,0 +1,68 @@
+        cout << out << endl;
+    }
+    return 0;
+}
```
Need a newline.

tiazahmd commented:
All of the following requested changes are made:
- test code with Test.h
- update README.md
- format code with clang-format
- check markdown documentions with markdownlint

MrHeer commented on .clang-format:
```diff
@@ -3,24 +3,29 @@ Language:        Cpp
 # BasedOnStyle:  LLVM
 AccessModifierOffset: -2
 AlignAfterOpenBracket: Align
+AlignConsecutiveMacros: false
```
Why update `.clang-format`?

MrHeer commented on src/PalindromeNumber/Solution.cpp:
```diff
@@ -0,0 +1,78 @@
+
+class Solution {
+public:
+  string isPalindrome(int x) {
```
Why return `string` not `bool`?

tiazahmd replied:
I started with `bool`, then changed it to `string` when I was testing with `Test.h`. No particular reason for changing it. Would you prefer `bool` or `string`?

MrHeer replied:
I think `bool` better.

MrHeer commented on src/PalindromeNumber/README.md:
```diff
@@ -0,0 +1,33 @@
+
+### Complexity Analysis
+
+* Time Complexity: `// TODO`
```
I think the time complexity of your algorithm is `O(log(n))`

tiazahmd replied:
Will make the change.

MrHeer commented on src/PalindromeNumber/README.md:
```diff
@@ -0,0 +1,33 @@
+### Complexity Analysis
+
+* Time Complexity: `// TODO`
+* Space Complexity: `// TODO`
```
I think the space complexity of your algorithm is `O(log(n))`

tiazahmd replied:
Will make the change.

MrHeer commented on src/PalindromeNumber/Solution.cpp:
```diff
@@ -0,0 +1,78 @@
+        vec.push_back(num);
+      }
+
+      for (int i = 0; i < vec.size(); i++) {
```
according your algorithm, `i` reaches the middle of the vector.

tiazahmd replied:
`i` is not supposed to go beyond half way. It compares the `first` and `last` integer, then the `(first + 1)`th and `(last - 1)`th digit and so on. In order to do that, the loop does not need to go beyond half way through.

MrHeer replied:
for  example, vec ＝ { 1, 2, 3, 2, 1}, i will to go 4.

tiazahmd replied:
So far, what I can see is that it is working as a "failsafe" so that if there are even number of integers, it will make sure that integer is accounted for. I can't seem to find a more elegant way to do this without compromising speed.

MrHeer replied:
@Zhongnibug What do you think？

MrHeer commented on src/PalindromeNumber/Solution.cpp:
```diff
@@ -0,0 +1,78 @@
+      long ct = 1;
+      int num = 0;
+      int digCount = 0;
+      int temp = x;
```
I think the `ct` and `digCount` are not necessary.
you can use a `while` loop to fill vector like this
```cpp
while (temp != 0) {
  num = temp % div;
  temp = temp / div;
  vec.push_back(num);
}
```

tiazahmd replied:
Will make the changes.

MrHeer commented on src/PalindromeNumber/Solution.cpp:
```diff
@@ -0,0 +1,78 @@
+
+  test("Test case: ", s.isPalindrome(121), "true");
+  test("Test case: ", s.isPalindrome(120), "true");
+  test("Test case: ", s.isPalindrome(-121), "true");
```
should return `false`

tiazahmd replied:
Will make the changes.

tiazahmd commented:
The following changes have been made to the new commit:
- Updated README.md with time and space complexity
- Function now returns `bool` instead of `string`
- Removed `ct` and `digCount` and made filling vector easier and faster
- Added one more test cases, with updated expectation and output & expectation type

MrHeer commented:
@tiazahmd Can you help update https://github.com/XFreeCoder/LeetCode/blob/master/README.md

Please reset `.clang-format`

tiazahmd commented:
Updated master `README` and reset `.clang-format`.

MrHeer commented on .clang-format:
```diff
@@ -111,4 +111,4 @@ SpacesInSquareBrackets: false
 TabWidth:        8
 UseTab:          Never
-...
+...
```
There should have a newline at end of file.

MrHeer commented on src/PalindromeNumber/Solution.cpp:
```diff
@@ -0,0 +1,70 @@
+  //   cout << out << endl;
+  // }
+  return 0;
+}
```
1. There should have a newline at end of file.
2. clean up the code.

Zhongnibug commented:
I think if you could define that a number,xr, which read forward,  equal to x reading backward,you will omit vector.And when xr is equal to x,x is a Palindrome Number.

MrHeer commented on .clang-format:
```diff
@@ -111,4 +111,5 @@ SpacesInSquareBrackets: false
 UseTab:          Never
-...
+
+...
```
This file should not be modified.

tiazahmd replied:
I've made sure the file is restored to its previous state.

MrHeer replied:
```bash
$ git diff master imtiaz-branch .clang-format
```
> diff --git a/.clang-format b/.clang-format
> index 4b76204..f0ca9d3 100644
> --- a/.clang-format
> +++ b/.clang-format
> @@ -111,4 +111,5 @@ SpacesInSquareBrackets: false
>  Standard:        Cpp11
>  TabWidth:        8
>  UseTab:          Never
> -...
> +
> +...
> \ No newline at end of file

tiazahmd commented:
I will be out of town until Tuesday. I’ll make the suggested changes once I’m back.

MrHeer commented on src/PalindromeNumber/Solution.cpp:
```diff
@@ -0,0 +1,57 @@
+  test("Test case: ", s.isPalindrome(0), true);
+  test("Test case: ", s.isPalindrome(10022001), true);
+  test("Test case: ", s.isPalindrome(1001), true);
+  
```
There should be no spaces here.

MrHeer replied:
```suggestion

```

MrHeer commented on src/PalindromeNumber/README.md:
```diff
@@ -0,0 +1,33 @@
+
+### AC Result
+
+`11509 / 11509 test cases passed`
```
In order to unify style, it is suggested to change `Inline code` to **Bold**.

tiazahmd commented:
Made suggested changes. I couldn't get @Zhongnibug 's solution yet. Let me know if you guys still want me to work on that, or if the current solution suffices.

MrHeer commented:
> Made suggested changes. I couldn't get @Zhongnibug 's solution yet. Let me know if you guys still want me to work on that, or if the current solution suffices.

Your solution is okey, but @Zhongnibug 's solution is better. Of course, for the best code, if you want to continue to optimize, you can refer to the [official solution](https://leetcode.com/articles/palindrome-number/#approach-1-revert-half-of-the-number)

MrHeer commented on src/PalindromeNumber/Solution.cpp:
```diff
@@ -0,0 +1,56 @@
+
+  Solution s;
+
+  test("Test case: ", s.isPalindrome(121), true);
+  test("Test case: ", s.isPalindrome(120), false);
+  test("Test case: ", s.isPalindrome(-121), false);
+  test("Test case: ", s.isPalindrome(11), true);
+  test("Test case: ", s.isPalindrome(9), true);
+  test("Test case: ", s.isPalindrome(0), true);
+  test("Test case: ", s.isPalindrome(10022001), true);
+  test("Test case: ", s.isPalindrome(1001), true);
```
```suggestion
  test("Test case 1", s.isPalindrome(121), true);
  test("Test case 2", s.isPalindrome(120), false);
  test("Test case 3", s.isPalindrome(-121), false);
  test("Test case 4", s.isPalindrome(11), true);
  test("Test case 5", s.isPalindrome(9), true);
  test("Test case 6", s.isPalindrome(0), true);
  test("Test case 7", s.isPalindrome(10022001), true);
  test("Test case 8", s.isPalindrome(1001), true);
```

tiazahmd commented:
> @tiazahmd Hello, Is there problem?

No problem. Is there something I have to do to finish this?

MrHeer commented:
@tiazahmd  I hope you can adopt my suggestion.
After update, you can `Resolve conversation`
